tests/test_notifications.py

```
@@ -0,0 +1,45 @@
+"""Tests for the notification service (ntfy auth behaviour, #90)."""
+from unittest.mock import patch
+
+from app.services.notifications import NotificationService
+
+
+class TestSendNtfy:
+    def test_refuses_token_over_plain_http(self, app):
+        """An access token must never be sent over an unencrypted connection."""
+        success, error = NotificationService.send_ntfy(
+            'http://ntfy.example.com/topic', 'Title', 'Message', token='tk_secret'
```

Ruff (0.16.0)

[error] Possible hardcoded password assigned to argument: "token"

(S106)